### PR TITLE
Autocomplete: Don't init when not logged in and add UI to show that to users

### LINKED ISSUE #928: Autocomplete: Improve sad users metrics
c.f. https://sourcegraph.slack.com/archives/C05AGQYD528/p1693471486690459

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #928

Avoid setting up autocomplete when the user is not signed in and log the states properly to the UI.

## Test plan

https://github.com/sourcegraph/cody/assets/458591/1444833c-0bce-4377-a894-cb97022ec2fa

### CODE REVIEW

**Review by toolmantim — APPROVED** [review]
Nice! 👍🏼 on the UX

**Review by dominiccooney — APPROVED** [review]
Looks good!

I think the AuthProvider publishes a variable, cody.activated, when it is authed. We could make some of these indicators declarative.

File: vscode/src/main.ts [hunk 1]
```diff
@@ -409,19 +431,24 @@ const register = async (
...
             contextProvider,
-            featureFlagProvider
-        )
+            featureFlagProvider,
+            authProvider,
+        })
     }
+    // Reload autocomplete if either the configuration changes or the auth status is updated
     vscode.workspace.onDidChangeConfiguration(event => {
         if (event.affectsConfiguration('cody.autocomplete')) {
```
[the remark this answers is not in the document]
Comment (philipp-spiess) [reply vscode/src/main.ts:445]: Oh I see this is using [this feature](https://code.visualstudio.com/api/extension-guides/command#using-a-custom-when-clause-context) but couldn't find out how we can listen to these changes in code.  I think this API is only for config changes but maybe I’m missing something? I'll merge this as-is but happy to change this if you find a nice way!

[the remark this answers is not in the document]
Comment (dominiccooney) [reply vscode/src/main.ts:445]: @philipp-spiess Yes, unfortunately you can't listen to these in code. But for making UX changes you can use `when`/`enablement` to do that declaratively.

**Pull request merged**